### ISSUE TITLE
Allow disabling or lowering type checking at file-level

### ISSUE THREAD

**Issue by Skylake-dev** [microsoft/pyright#3608]
It possible to elevate the level on a file by using comments as stated [here](https://github.com/microsoft/pyright/blob/main/docs/comments.md#file-level-type-controls):

_Strict type checking, where most supported type-checking switches generate errors, can be enabled for a file through the use of a special comment. Typically this comment is placed at or near the top of a code file on its own line._
`# pyright: strict`

Now it is not possible to do the opposite. For instance i would like to disable type checking on a specific file and leave it enabled for all the other files in my project.
The ideal solution would be, instead of having to add `# type: ignore` on every problematic line, to add a comment like `# pyright: off` or `# pyright: basic` at the top of the file and leaving the `"python.analysis.typeCheckingMode": "strict",` line in the settings.json file. I tried to do so but it doesn't seem to be possible.

**Comment by erictraut** [comment]
You can override specific rules in a file that is otherwise strict.

```python
# pyright: reportIncompatibleVariableOverride=false
```

The thinking is that if you have opted in to strict type checking for a file or a directory, you should be very specific when overriding those rules. If you want to exclude a file or directory from strict mode, you should not include it in the `"strict"` configuration setting to begin with.

The standard approach I used in my team's code base was to start with basic type checking for all of the files in a directory. As I made each file strictly typed, I added `# pyright: strict` to the top of the file to prevent regressions. Once all of the files in that directory reached "strict" status, I added the entire directory to the "strict" setting in the config file and deleted the individual `# pyright: strict` comments in those files.

Pyright composes diagnostic severity settings from many sources ("typeCheckingMode" from user settings, "typeCheckingMode" from project settings, "typeCheckingMode" from config file, individual severity overrides for all of the above, "strict" mode overrides in the config file, and per-file comments). It's complex and delicate logic to get these to compose and override correctly. Allowing `# pyright: basic` to override "strict" from the config file would likely create problems, so I'm not inclined to change this.

**Comment by erictraut** [comment]
Yes, those are two options. Another is to do what I mentioned above and leave the file as strict but use file-level `# pyright: ` comments to disable specific rules.

**Comment by Skylake-dev** [comment]
Yes of course.
Thank you for your time.
